ledger/participant-integration-api/src/main/scala/platform/store/backend/common/UserManagementStorageBackendTemplate.scala

```diff
@@ -70,18 +77,30 @@ object UserManagementStorageBackendTemplate extends UserManagementStorageBackend
       }
   }
 
-  override def getUsers()(connection: Connection): Vector[domain.User] = {
-    def domainUser(userId: String, primaryParty: Option[String]): domain.User = {
-      domain.User(
-        Ref.UserId.assertFromString(userId),
-        primaryParty.map(Ref.Party.assertFromString),
-      )
-    }
-    SQL"""SELECT internal_id, user_id, primary_party
-          FROM participant_users"""
+  override def getUsersOrderedById(after: UserId, maxResults: Int)(
+      connection: Connection
+  ): Vector[domain.User] = {
+    import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
+    SQL"""SELECT user_id, primary_party
+          FROM participant_users
+          WHERE user_id > ${after: String}
+          ORDER BY user_id
+          ${limitClause(number = maxResults)}"""
       .asVectorOf(ParticipantUserParser2)(connection)
       .map { case (userId, primaryPartyRaw) =>
-        domainUser(userId, dbStringToPartyString(primaryPartyRaw))
+        toDomainUser(userId, dbStringToPartyString(primaryPartyRaw))
+      }
+  }
+
+  override def getUsersOrderedById(maxResults: Int)(connection: Connection): Vector[domain.User] = {
+    import com.daml.platform.store.backend.common.ComposableQuery.SqlStringInterpolation
+    SQL"""SELECT user_id, primary_party
+          FROM participant_users
+          ORDER BY user_id
```

Comment:
This query could be unified with the above one by exploiting that "" is never a valid user_id; i.e., using that one as the lower bound for the first page works.

Reply:
Ok, adopting single method approach